bson/fuzz_test.go

```diff
@@ -0,0 +1,34 @@
+package bson
+
+import (
+	"testing"
+)
+
+func FuzzDecode(f *testing.F) {
+	seedBSONCorpus(f)
+
+	f.Fuzz(func(t *testing.T, data []byte) {
+		for _, typ := range []func() interface{}{
```

Comment:
Why not loop over `[]interface{}`? Is there a reason I'm not seeing for using these constructors?

Reply:
[It's a practice I took from the Go team](https://cs.opensource.google/go/go/+/master:src/encoding/json/fuzz_test.go;l=30). I think the idea is to make the type instantiation more realistic,  i.e. something like this:

```go
x := interface{} // or in our case typ()
// unmarshal into &x
```

And not like
```go
for _, x := range []interface{ ... } {
// unmarshal into &x
}
```